dap-mode.el

```diff
@@ -611,15 +611,18 @@ thread exection but the server will log message."
   "Make STACK-FRAME the active STACK-FRAME of DEBUG-SESSION."
   (let ((lsp--cur-workspace (dap--debug-session-workspace debug-session)))
     (when stack-frame
-      (-let (((&hash "line" line "column" column "name" name) stack-frame)
-             (source (gethash "source" stack-frame)))
-        (if source
-            (progn (find-file (gethash "path" source))
-                   (setf (dap--debug-session-active-frame debug-session) stack-frame)
-
+      (-let* (((&hash "line" line "column" column "name" name) stack-frame)
+              (source (gethash "source" stack-frame))
+              (path (gethash "path" source)))
+        (setf (dap--debug-session-active-frame debug-session) stack-frame)
+        ;; If we have a source file with path attached, open it and
+        ;; position the point in the line/column referenced in the
+        ;; stack trace.
+        (if (and source path)
```

Comment:
at this point, it is impossible to have source=nil since we have already executed `(path (gethash "path" source))`

Reply:
I think that this line

> (path (gethash "path" source))

should be changed to 

> (path (and source (gethash "path" source)))